main.py

```diff
@@ -12,9 +12,68 @@
 # Radius of earth in kilometers.
 EARTH_RADIUS = 6371
 
+# If set to TRUE, results will be exported to google maps
+GOOGLE_MAPS_EXPORT = False
+
 # Max distance you want to travel, in kilometers
 MAX_DISTANCE = 2000
 
+def main():
+    start_time = time.perf_counter()
+    lat = start_lat
+    lon = start_lon
+    distances_df = pd.DataFrame()
+    travel_df = pd.DataFrame(
+        columns=[
+            "brewery_id",
+            "distance",
+            "name",
+            "latitude",
+            "longitude",
+            "beer_count",
+            "distance_to_home",
+            "ratio",
+        ],
+        data=[[0, 0, "HOME", start_lat, start_lon, 0, 0, 0]],
+    )
+
+    while True:
+
+        # Apply the distance calculation formula
+        distances_df = (
+            geocodes_df.query("Visited == False")
+            .apply(apply_distance_calc, args=(lat, lon), axis=1)
+            .rename("distance")
+            .to_frame()
+        )
+        # Join additional data from geocodes to distances_df
+        distances_df = distances_df.join(
+            geocodes_df[["name", "latitude", "longitude", "beer_count"]],
+            on="brewery_id",
+            how="inner",
+        )
+        # Calculate how many km you can still travel and get next valid location
+        km_left = MAX_DISTANCE - travel_df.distance.sum()
+        next_valid_location = get_next_location(distances_df, km_left)
+
+        if next_valid_location.empty:
+            # Add HOME location to the end of the dataframe
+            travel_df = travel_df.append(travel_df.iloc[0])
+            travel_df.iat[-1, 0] = travel_df.tail(2).distance_to_home.values[0]
+            travel_df = travel_df.rename_axis("brewery_id").reset_index()
```

Comment:
I think this part should be nicer.
I understand you have to home as your last destination. Are there alternative ways?

Reply:
Changed